src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -39,10 +47,12 @@ static IamResourceType fromValue(String text) {
   }
 
   public static IamResourceType fromEnum(IamResourceTypeEnum apiEnum) {
-    if (apiEnum == IamResourceTypeEnum.SPEND_PROFILE) {
-      return IamResourceType.SPEND_PROFILE;
+    for (IamResourceType b : IamResourceType.values()) {
```

Comment:
You could stream this
```
   return Arrays.stream(values()).filter(type.iamResourceTypeEnum == apiEnum).findFirst().orElse(null);
```

Reply:
I took that a step further! I would love some eyes, I haven't done as much functional interface stuff in Java so I'm a bit more shaky on that.